features/step_definitions/operators.rb

```diff
@@ -160,3 +160,34 @@
   cb_name = 'operator_channel_name' unless cb_name
   cb[cb_name] = cluster_version('version').channel.split('-')[1]
 end
+
+Given /^I wait(?: up to (\d+) seconds)? for operator "([^"]*)" to become conditions: "([^"]*)"$/ do | time, operator, conditions |
+  ensure_admin_tagged
+
+  expected = {}
+  # Parse the conditions to Hash value {Available"=>"True", "Progressing"=>"False", "Degraded"=>"False", "Upgradeable"=>"True"}
+  arr_conditions = conditions.split(/\s+/)    
+  raise "#### Invalid number of conditions!!! (Valid number:1~4)" unless arr_conditions.length <=4
+  if arr_conditions.any? { |x| x.include? "=" }
+    arr_conditions.map{|ar| ar.split('=')}.each {|k, v| expected[k]=v}
+  else
+    # Parse the conditions with "True False ..." input 
+    ["Available", "Progressing", "Degraded", "Upgradeable"][0,arr_conditions.length].zip(arr_conditions).map {|k, v| expected[k]=v}
+  end
+  expected.each {|k,v| raise "#### Please make sure the input status of condition is True or False!" unless ["True","False"].include? v }
+  raise "#### Without valid conditions, unable to continue to check !!!" unless expected.length >0
+
+  interval_time = 5 
+  time = Integer(time) rescue 60
+  interval_time = 20 if time > 100
+  actual_results = {}
+  success = wait_for(time, interval: interval_time) {
+    expected.keys.each do |c|
+        actual_results[c] = cluster_operator(operator).condition(type: c, cached: false)['status']
+    end
+    expected == actual_results
+  }
+  logger.info("#### Operator #{operator} Expected conditions: #{expected}")
+  logger.info("#### Operator #{operator} Actual conditions: #{actual_results}")
+  raise "The #{operator} operator still didn't become ready after #{time} seconds" unless success
```

Comment:
Not "`become ready`". Change to `become #{expected}`

Reply:
agree.